ReactSkia/views/common/RSkDrawUtils.cpp

```diff
@@ -74,29 +76,39 @@ void setStyle(int strokeWidth,SkPaint::Style style,BorderStyle borderStyle,SkPai
         setPathEffect(borderStyle,strokeWidth,paint);
     }
 }
-void setColor(SharedColor Color,SkPaint *paint)
+inline bool isBorderColorVisible(SharedColor Color)
 {
-    paint->setAntiAlias(true);
-    paint->setColor(RSkColorFromSharedColor(Color, DEFAULT_COLOUR));
-}
-inline bool isColorVisible(SharedColor Color){
-    return (Color && colorComponentsFromColor(Color).alpha) ? true:false;
+    return ((!Color) || //default color is fully opaque black for Border,when color not specifed, so returning true
+            (colorComponentsFromColor(Color).alpha != 0.0)
+            ) ? true:false;
 }
 inline bool isBorderEdgeVisible(SharedColor Color,Float thickness=1.0)
 {
-    return (isColorVisible(Color) && thickness)? true:false;
+
+//Note::Default color for border is fully opaque black, When Color not Specifeid but has visible width/thickness
+
+    return (thickness &&
+            isBorderColorVisible(Color)
+           )? true:false;
 }
 inline bool hasVisibleBorder(BorderColors borderColor,BorderWidths borderWidth)
 {
-    /*Returns true, when atleast one side with  visible colour & thickness*/
-    if((borderWidth.left || borderWidth.right|| borderWidth.top|| borderWidth.bottom) &&
-       (isColorVisible(borderColor.left) || isColorVisible(borderColor.right)||
-        isColorVisible(borderColor.top)|| isColorVisible(borderColor.bottom)))
-    {
+    /*Returns true, when atleast one side with visible thickness & color*/
+
+    if(isBorderEdgeVisible(borderColor.left,borderWidth.left) ||
+       isBorderEdgeVisible(borderColor.right,borderWidth.right) ||
+       isBorderEdgeVisible(borderColor.top,borderWidth.top) ||
+       isBorderEdgeVisible(borderColor.bottom,borderWidth.bottom)) {
         return true;
     }
     return false;
 }
+inline bool hasVisibleBackGround(SharedColor Color){
```

Comment:
can we make use of framework provided API isColorMeaningful() which does the same .

Reply:
As we discussed and concluded, we will continue here as is isColorMeaningful() is in  viewShadoeNode. and to avoid file inclusion for just an API.